### PR TITLE
xds-k8s: Adapt URL Map tests to use resource prefix/suffix

### PR DESCRIPTION
**Pull request by sergiitk**
1. fixes `TypeError: __init__() missing 1 required keyword-only argument: 'resource_suffix'` caused by https://github.com/grpc/grpc/pull/26542
2. isort
3. minor pylint fixes/suggestions

Test run: https://source.cloud.google.com/results/invocations/dd413577-cc68-4e6e-843b-27e1eb598ffe

### CODE REVIEW

File: tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py [hunk 1]
```diff
@@ -132,16 +132,22 @@ class GcpResourceManager(metaclass=_MetaSingletonAndAbslFlags):
...
+            for key in absl_flags:
+                setattr(self, key, absl_flags[key])
         # API managers
         self.k8s_api_manager = k8s.KubernetesApiManager(self.kube_context)
         self.gcp_api_manager = gcp.api.GcpApiManager()
         self.td = traffic_director.TrafficDirectorManager(
             self.gcp_api_manager,
             self.project,
+            # TODO(lidiz): replace namespace with resource prefix/suffix.
             resource_prefix=self.namespace,
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py:149]: We can just use `resource_prefix=self.resource_prefix`.

File: tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py [hunk 2]
```diff
@@ -132,16 +132,22 @@ class GcpResourceManager(metaclass=_MetaSingletonAndAbslFlags):
...
+            resource_suffix="",
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py:150]: `resource_suffix=self.resource_suffix`

**Review by lidizheng — COMMENTED** [review]
Thanks for the fix. Can you try out the suggested changes? It might save another round of PR/review.

File: tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py (hunk 1 above)
Comment (sergiitk) [reply tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py:149]: To make this work, we need to update this as well: https://github.com/grpc/grpc/blob/ee4f6854bd6d56a16187923421c940e5a47a5a49/tools/internal_ci/linux/grpc_xds_url_map.sh#L96

**Review by lidizheng — APPROVED** [review]
Thanks for the change! Appreciate the quick fix!

File: tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py (hunk 2 above)
Comment (sergiitk) [reply tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_test_resources.py:150]: `self.resource_suffix` is assumed to be handled by the test case, so it's None by default.
I can make it `resource_suffix="" if self.resource_suffix is None else self.resource_suffix`

**Comment by sergiitk** [comment]
Last Check log in ResultStore UI: https://source.cloud.google.com/results/invocations/34744e43-869b-448d-9f53-8fd9375abf59

**Pull request merged**